lib/runner/variant-reassignment.js

```diff
@@ -1,19 +1,28 @@
+import ProductService from '../services/product-manager'
+import TransactionService from '../services/transaction-manager'
+
 export default class VariantReassignment {
 
-  constructor (logger, options, blackList) {
+  constructor (client, logger, options, blackList) {
     this.unfinishedTransactions = []
     this.firstRun = true
     this.customObjectService = null // build custom object service
     this.blackList = blackList
     this.options = options
     this.logger = logger
+    this.productService = new ProductService(logger, client)
+    this.transactionService = new TransactionService(logger, client)
   }
 
   execute (productDrafts, existingProducts) {
     this._processUnfinishedTransactions()
 
-    for (const productDraft of productDrafts)
-      this._processProductDraft(productDraft, existingProducts)
+    const productDraftsForReassignment
+      = this._selectProductDraftsForReassignment(productDrafts, existingProducts)
+
+    if (productDraftsForReassignment.length)
+      for (const productDraft of productDraftsForReassignment)
+        this._processProductDraft(productDraft)
```

Comment:
add await

Reply:
So far I don't need it. Let's add once it's necessary, ok?